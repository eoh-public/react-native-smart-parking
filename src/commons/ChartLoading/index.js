import React from 'react';
import Svg, { Text } from 'react-native-svg';
import t from 'i18n';

const ChartLoading = ({ message }) => {
  return (
    <Svg width={'100%'} height={300}>
      <Text
        x="50%"
        y="47%"
        alignment-baseline="middle"
        textAnchor="middle"
        fontSize="20"
        fontWeight="bold"
        fill="black"
      >
        {message || t('loading')}
      </Text>
    </Svg>
  );
};
export default ChartLoading;
