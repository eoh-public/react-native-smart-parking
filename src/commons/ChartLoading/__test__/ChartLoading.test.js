import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Text } from 'react-native-svg';
import ChartLoading from '../index';

describe('Test chart loading', () => {
  let tree;
  test('create chart loading', () => {
    act(() => {
      tree = renderer.create(<ChartLoading message="loading" />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('create chart  message null', () => {
    act(() => {
      tree = renderer.create(<ChartLoading />);
    });
    const instance = tree.root;
    const textInputs = instance.findAllByType(Text);
    expect(textInputs.length).toBe(1);
  });
});
