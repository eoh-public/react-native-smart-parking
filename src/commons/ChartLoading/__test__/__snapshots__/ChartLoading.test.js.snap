// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test chart loading create chart loading 1`] = `
<RNSVGSvgView
  bbHeight={300}
  bbWidth="100%"
  focusable={false}
  height={300}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 300,
        "width": "100%",
      },
    ]
  }
  width="100%"
>
  <RNSVGGroup>
    <RNSVGText
      content={null}
      dx={Array []}
      dy={Array []}
      fill={4278190080}
      font={
        Object {
          "fontSize": "20",
          "fontWeight": "bold",
          "textAnchor": "middle",
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      rotate={Array []}
      x={
        Array [
          "50%",
        ]
      }
      y={
        Array [
          "47%",
        ]
      }
    >
      <RNSVGTSpan
        content="loading"
        dx={Array []}
        dy={Array []}
        font={Object {}}
        rotate={Array []}
        x={Array []}
        y={Array []}
      />
    </RNSVGText>
  </RNSVGGroup>
</RNSVGSvgView>
`;
